File: tarea-2/Program/Items/Item.java
package Program.Items;

import Program.Mascota;

public abstract class Item {
    protected int id;
    protected String nombre;
    protected int cantidad; 

    public Item(int id, String nombre, int cantidad){
        this.id = id;
        this.nombre = nombre;
        this.cantidad = cantidad;
    }

    // metodo usarItem se encuentra definido en cada subclase de Item (Juguete, Medicina, Comida)
    // aumenta las estadisticas respectivas de la mascota y consume item (a menos que sea Juguete)
    public abstract void usarItem(Mascota mascota);



    //GETers & SETers
    public int getId() {return id;}
    public void setId(int newId) {id = newId;}

    public String getNombre() {return nombre;}
    public void setNombre(String newNombre) {nombre = newNombre;}

    public int getCantidad() {return cantidad;}
    public void setCantidad(int newCantidad) {cantidad = newCantidad;} 

}

File: tarea-3/etapa-2/Item.cpp
#include <iostream>
#include <string>
#include "../headers/Item.h"
#include "../headers/Mascota.h"

Item::Item(int i, std::string t, std::string n, int c) {
	id = i;
	tipo = t;
	nombre = n;
	cantidad = c;
}
int Item::getId() const {
	return id;
}
void Item::usarItem(Mascota &mascota) {
	std::cout << "Usando " << tipo << " " << nombre << " en la Mascota" << std::endl;
	mascota.mostrarMascota();
} 
std::string Item::getTipo() {
	return tipo;
}
std::string Item::getNombre() {
	return nombre;
}
int Item::getCant() {
	return cantidad;
}
void Item::setCant(int newCant) {
	cantidad = newCant;
}
void Item::mostrarItem() const {
	std::cout << "ID: " << id << ", Nombre: " << nombre << ", Cantidad " << cantidad << std::endl;
};
